server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/QProfilesWsMediumTest.java

```diff
@@ -396,6 +399,142 @@ public void reset() throws Exception {
+      .execute();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void add_project_missing_project() throws Exception {
+    wsTester.newPostRequest(QProfilesWs.API_ENDPOINT, "add_project")
+      .setParam("profileKey", "plouf")
+      .execute();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void add_project_too_many_project_parameters() throws Exception {
+    wsTester.newPostRequest(QProfilesWs.API_ENDPOINT, "add_project")
+      .setParam("profileKey", "plouf").setParam("projectUuid", "polop").setParam("projectKey", "palap")
+      .execute();
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void add_project_unknown_profile() throws Exception {
+    wsTester.newPostRequest(QProfilesWs.API_ENDPOINT, "add_project")
+      .setParam("projectUuid", "plouf").setParam("profileName", "polop").setParam("language", "xoo")
+      .execute();
+  }
+
+  @Test
+  public void remove_project_with_key_and_uuid() throws Exception {
+    ComponentDto project = new ComponentDto()
+      .setId(1L)
+      .setUuid("ABCD")
+      .setKey("org.codehaus.sonar:sonar")
+      .setName("SonarQube")
+      .setLongName("SonarQube")
+      .setQualifier("TRK")
+      .setScope("TRK")
+      .setEnabled(true);
+    db.componentDao().insert(session, project);
+    QualityProfileDto profile = QProfileTesting.newXooP1();
+    db.qualityProfileDao().insert(session, profile);
+    db.qualityProfileDao().insertProjectProfileAssociation(project.uuid(), profile.getKee(), session);
+
+    session.commit();
+
+    wsTester.newPostRequest(QProfilesWs.API_ENDPOINT, "remove_project")
+      .setParam("profileKey", profile.getKee()).setParam("projectUuid", project.uuid())
+      .execute().assertNoContent();
+    assertThat(tester.get(QProfileFactory.class).getByProjectAndLanguage(session, project.getKey(), "xoo")).isNull();
+  }
+
+  @Test
+  public void remove_project_with_name_language_and_key() throws Exception {
+    ComponentDto project = new ComponentDto()
+      .setId(1L)
+      .setUuid("ABCD")
+      .setKey("org.codehaus.sonar:sonar")
+      .setName("SonarQube")
+      .setLongName("SonarQube")
+      .setQualifier("TRK")
+      .setScope("TRK")
+      .setEnabled(true);
+    db.componentDao().insert(session, project);
+    QualityProfileDto profile = QProfileTesting.newXooP1();
+    db.qualityProfileDao().insert(session, profile);
+    db.qualityProfileDao().insertProjectProfileAssociation(project.uuid(), profile.getKee(), session);
+
+    session.commit();
+
+    wsTester.newPostRequest(QProfilesWs.API_ENDPOINT, "remove_project")
+      .setParam("language", "xoo").setParam("profileName", profile.getName()).setParam("projectKey", project.getKey())
+      .execute().assertNoContent();
+    assertThat(tester.get(QProfileFactory.class).getByProjectAndLanguage(session, project.getKey(), "xoo")).isNull();
+  }
+
```

Comment:
These test case should be moved in separated tests using only DbTester and DbTester.